data/sea/52-psv.h

```diff
@@ -142,10 +148,9 @@ static ierror_msg_t psv_read_buffer (psv_state_t *s)
         const char  *value_end  = time_ptr - 1;
         const size_t value_size = value_end - value_ptr;
 
-        const bool new_entity = entity_cur_size != entity_size
-                             || memcmp (entity_cur, entity_ptr, entity_size) != 0;
+        const int new_entity = memcmp (entity_cur, entity_ptr, entity_size);
```

Comment:
Yeah you're right this is broken, although `entity_size` refers to `entity_ptr` so it would compare `AB` to `AB`.